rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/server/RMServiceImpl.java

```diff
@@ -288,222 +258,146 @@ public String defineNodeSource(final String sessionId, final String nodeSourceNa
-            final List<String> attrs, final String timeRange) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getNodeMBeansHistory(sessionId, nodeJmxUrl, objectNames, attrs, timeRange);
-            }
-        });
+    public String getNodeMBeansHistory(String sessionId, String nodeJmxUrl, String objectNames, List<String> attrs,
+            String timeRange) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getNodeMBeansHistory(sessionId,
+                                                                                                 nodeJmxUrl,
+                                                                                                 objectNames,
+                                                                                                 attrs,
+                                                                                                 timeRange));
     }
 
     @Override
-    public String getStatHistory(final String sessionId, final String range)
-            throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.getStatHistory(sessionId, range);
-            }
-        });
+    public String getStatHistory(String sessionId, String range) throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.getStatHistory(sessionId, range));
     }
 
     @Override
-    public String executeNodeScript(final String sessionId, final String script, final String engine,
-            final String nodeUrl) throws RestServerException, ServiceException {
-        return executeFunctionReturnStreamAsString(new Function<RestClient, InputStream>() {
-            @Override
-            public InputStream apply(RestClient restClient) {
-                return restClient.executeNodeScript(sessionId, nodeUrl, script, engine);
-            }
-        });
+    public String executeNodeScript(String sessionId, String script, String engine, String nodeUrl)
+            throws RestServerException, ServiceException {
+        return executeFunctionReturnStreamAsString(restClient -> restClient.executeNodeScript(sessionId,
+                                                                                              nodeUrl,
+                                                                                              script,
+                                                                                              engine));
     }
 
     private RestClient getRestClientProxy() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)